testing/database.rst

```diff
@@ -4,40 +4,127 @@
------------------------------------------
+The above assumes that each developer/machine uses a different database for the
+tests. If the entire team uses the same settings for tests, edit or create the
+``.env.test`` file instead and commit it to the shared repository. Learn more
+about :ref:`using multiple .env files in Symfony applications <configuration-multiple-env-files>`.
 
-If you want to test code which depends on a Doctrine repository in isolation,
-you need to mock the ``Repository``. Normally you inject the ``EntityManager``
-into your class and use it to get the repository. This makes things a little
-more difficult as you need to mock both the ``EntityManager`` and your repository
-class.
+Resetting the Database Automatically Before each Test
+-----------------------------------------------------
 
-.. tip::
+Tests should be independent from each other to avoid side effects. However, if
+some test modifies the database (for example by adding or removing an element)
+it could impact the other tests. Run the following command to install a bundle
+that ensures that each test is run with a new and unmodified copy of the database:
+
+.. code-block:: terminal
+
+    $ composer require --dev dama/doctrine-test-bundle
+
+Now, enable it as a PHPUnit extension (PHPUnit 8 or higher) or a listener
+(PHPUnit 7 or lower):
+
+.. code-block:: xml
+
+    <!-- phpunit.xml.dist -->
+    <phpunit>
+        <!-- ... -->
+
+        <!-- Add this in PHPUnit 8 or higher -->
+        <extensions>
+            <extension class="DAMA\DoctrineTestBundle\PHPUnit\PHPUnitExtension" />
+        </extensions>
+
+        <!-- Add this in PHPUnit 7 or lower -->
+        <listeners>
+            <listener class="\DAMA\DoctrineTestBundle\PHPUnit\PHPUnitListener" />
+        </listeners>
+    </phpunit>
+
+This bundle uses a clever trick to create a new database for each test without
+scarifying performance: it begins a transaction before every test case and rolls
+it back automatically after the test finished. Read more in the documentation of
+the `DAMADoctrineTestBundle`_.
+
+.. _doctrine-fixtures:
 
-    It is possible (and a good idea) to inject your repository directly by
-    registering your repository as a :doc:`factory service </service_container/factories>`.
-    This is a little bit more work to setup, but makes testing easier as you
-    only need to mock the repository.
+Dummy Data Fixtures
+-------------------
+
+Instead of using the real data from the production database, it's common to use
+fake or dummy data in the test database. This is usually called *"fixtures data"*
+and Doctrine provides a library to create and load them. Install it with:
+
+.. code-block:: terminal
+
+    $ composer require --dev doctrine/doctrine-fixtures-bundle
+
+Then, use the ``make:fixtures`` command to generate an empty fixture class:
+
+.. code-block:: terminal
+
+    $ php bin/console make:fixtures
+
+    The class name of the fixtures to create (e.g. AppFixtures):
```

Comment:
is the last `s` of `AppFixtures` an error? or from the maker code?
it is confusing at first, moreover if next line you enter `ProductFixture` without s

Reply:
Yes, Maker bundle uses "AppFixtures" (https://github.com/symfony/maker-bundle/blob/020c04826455abf72184d487a1da65a2632f5fd3/src/Maker/MakeFixtures.php)